examples/embeddable_examples/public/react_embeddables/search/search_react_embeddable.tsx

```diff
@@ -24,54 +26,39 @@ export const getSearchEmbeddableFactory = (services: Services) => {
     deserializeState: (state) => {
       return state.rawState as State;
     },
-    buildEmbeddable: async (state, buildApi) => {
+    buildEmbeddable: async (state, buildApi, uuid, parentApi) => {
+      const {
+        appliedTimeRange$,
+        cleanupTimeRange,
+        serializeTimeRange,
+        timeRangeApi,
+        timeRangeComparators,
+      } = initializeTimeRange(state, parentApi as Partial<PublishesTimeRange>);
```

Comment:
nit: can we make this initializer take `unknown` as the parent type and just check the parent against `apiPublishesTimeRange` internally? That way we can avoid the cast on the consumer side.